db/db_compaction_test.cc

```diff
@@ -9890,6 +9890,76 @@ TEST_F(DBCompactionTest, TurnOnLevelCompactionDynamicLevelBytesUCToLC) {
   ASSERT_EQ(expected_lsm, FilesPerLevel(1));
 }
 
+TEST_F(DBCompactionTest, RefitFilesFromNonL0ToL0) {
+  Options options = CurrentOptions();
+  options.compaction_style = CompactionStyle::kCompactionStyleLevel;
+  options.num_levels = 3;
+  std::shared_ptr<SstPartitionerFactory> factory(
+      NewSstPartitionerFixedPrefixFactory(1));
+  options.sst_partitioner_factory = factory;
+  DestroyAndReopen(options);
+
+  // To set up LSM shape:
+  // L0
+  // L1
+  // L2: ["a"], ["b"], ["c"] (sorted by ascending smallest key)
+  // All of these three files have epoch number = 1
+  ASSERT_OK(Put("a", "v"));
+  ASSERT_OK(Flush());
+
+  ASSERT_OK(Put("a", "v"));
+  ASSERT_OK(Put("c", "v"));
+  const Snapshot* s1 = db_->GetSnapshot();
```

Comment:
Do we need the snapshots and the multiple Puts for the same key?

Reply:
It was my dumb way of forcing no trivial move of these three files to L2 (in order to achieve same epoch number). I revised it based on your corruption test.

Reply:
ah - I didn't use the "do a kForce manual compaction" trick before. Should work. I am good with keeping the revised UT too.